### PR TITLE
[cmake] Unify usage of CMAKE_CROSSCOMPILE throughout the project

### PR DESCRIPTION
**Pull request by fetzerch**
## Description
With @wsnipex we decided to set CMAKE_CROSSCOMPILING always when we use a Toolchainfile (and build with depends). This was so far not the case on OSX. In order to allow this, we introduce another variable CORE_HOST_IS_TARGET that determines if we can run build artefacts such as unit tests on the build host.

+includes some cleanups of the toolchain files.

It's also worth mentioning, that with this change, we can build cover coverage of the unit tests on linux on jenkins.

## Motivation and Context
Consistency, and fixes a bug that currently on jenkins `make test` will not work because because CMAKE_CROSSCOMPILING is set and hence testing related targets are not created.

## How Has This Been Tested?
Will need jenkins for testing.

## Types of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by wsnipex** [comment]
looks good
jenkins build this with addons please

**Comment by fetzerch** [comment]
jenkins build this with addons please

**Comment by fetzerch** [comment]
jenkins build this with addons please

File: project/cmake/modules/FindPython.cmake [hunk 1]
```diff
@@ -5,7 +5,7 @@
 # PYTHON_INCLUDE_DIRS - the python include directory
 # PYTHON_LIBRARIES - The python libraries
 
-if(PKG_CONFIG_FOUND AND NOT CMAKE_CROSSCOMPILING)
```
Comment (fetzerch) [inline project/cmake/modules/FindPython.cmake]: @wsnipex: Are you familiar with the constrains of this module? (going back in the history i could see that you added the NOT CMAKE_CROSSCOMPILING here). according to the build log, i'd also have to exclude linux here now. which basically means we only use pkgconfig on osx.

**Comment by fetzerch** [comment]
jenkins build this with addons please

**Comment by fetzerch** [comment]
d359195 restores the behavior we had before, the FindPython module needs more love, but that can happen independently.

jenkins build this with addons please

**Comment by fetzerch** [comment]
@wsnipex: good to go?

File: project/cmake/modules/FindPython.cmake [hunk 2]
```diff
@@ -5,7 +5,7 @@
...
+if(PKG_CONFIG_FOUND AND (NOT CMAKE_CROSSCOMPILING OR APPLE))
```
Comment (wsnipex) [inline project/cmake/modules/FindPython.cmake]: what I was trying to say above is that _all platforms_ but windows should work with pkgconfig.

**Comment by wsnipex** [comment]
can we get this in? We can fix the findpython module later.

**Comment by fetzerch** [comment]
yes, fine by me

**Pull request merged**